src/ucp/wireup/address.c

```diff
@@ -669,37 +677,49 @@ ucp_address_pack_iface_attr_v1(ucp_worker_h worker, void *ptr,
     return sizeof(*packed);
 }
 
+size_t ucp_address_iface_seg_size(const uct_iface_attr_t *iface_attr)
+{
+    /* To be replaced by iface_attr.cap.am.max_recv when it is added to the
+     * UCT API */
+    if (iface_attr->cap.flags & UCT_IFACE_FLAG_AM_BCOPY) {
+        return iface_attr->cap.am.max_bcopy;
+    } else if (iface_attr->cap.flags & UCT_IFACE_FLAG_AM_ZCOPY) {
+       return iface_attr->cap.am.max_zcopy;
+    } else if (iface_attr->cap.flags & UCT_IFACE_FLAG_AM_SHORT) {
+        return iface_attr->cap.am.max_short;
+    } else {
+        return 0ul;
+    }
+}
+
 static unsigned
 ucp_address_pack_iface_attr_v2(ucp_worker_h worker, void *ptr,
                                const uct_iface_attr_t *iface_attr,
                                unsigned atomic_flags)
 {
-    unsigned seg_size                          = 0;
     ucp_address_v2_packed_iface_attr_t *packed = ptr;
     uint64_t addr_iface_flags;
+    double latency_nsec, overhead_nsec;
+    size_t seg_size;
 
-    packed->overhead  = UCS_FP8_PACK(OVERHEAD, iface_attr->overhead);
+    latency_nsec  = ucp_tl_iface_latency(worker->context, &iface_attr->latency) *
```

Comment:
why it worth packing `nsec`?
it seems that this change might make the current UCX version incompatible with the previous one.

Reply:
that's how we define `OVERHEAD` fp8 type in ucp_types.h. We unpack it to seconds. What do you mean by incompatible? It is for addr v2 only

Reply:
Sorry, I've noticed that this is for v2